lib/main.js

```diff
@@ -1,34 +1,36 @@
 'use strict'
 
 var dotenvExpand = function (config) {
-  var interpolate = function (env) {
-    var matches = env.match(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g) || []
-
-    matches.forEach(function (match) {
-      var key = match.replace(/\$|{|}/g, '')
-
-      // process.env value 'wins' over .env file's value
-      var variable = process.env.hasOwnProperty(key) ? process.env[key] : (config.parsed[key] || '')
-
-      // Resolve recursive interpolations
-      variable = interpolate(variable)
-
-      env = env.replace(match, variable)
-    })
-
-    return env
+  var interpolate = function (envValue) {
+    var matches = envValue.match(/(.?\${?(?:[a-zA-Z0-9_]+)?}?)/g) || []
+
+    return matches.reduce(function (newEnv, match) {
+      var parts = /(.?)\${?([a-zA-Z0-9_]+)?}?/g.exec(match)
+      var $prefix = parts[1]
```

Comment:
Would this just be better as `prefix` instead of prefixing it with a `$`, too?

Reply:
It is the prefix of the $ sign :), but I can change it if you insist